extensions/libarchive/ReadArchiveStream.cpp

```diff
@@ -0,0 +1,82 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ReadArchiveStream.h"
+
+namespace org::apache::nifi::minifi::io {
+
+ReadArchiveStreamImpl::archive_ptr ReadArchiveStreamImpl::createReadArchive() {
+  archive_ptr arch = archive_read_new();
+  if (!arch) {
+    logger_->log_error("Failed to create read archive");
+    return nullptr;
+  }
+
+  int result;
+
+  result = archive_read_support_format_all(arch.get());
```

Comment:
since this variable is reused for each operation I was reluctant to merge them, since that made me think that the result belongs to this operation only, I am not sure if that was only me or not, should we merge them?

Reply:
I didn't think about it like this, but if you think it improves the representation of this value and gives context to you or anyone else reading this, I don't insist on merging it.